client_test.go

```diff
@@ -228,3 +228,14 @@ func TestClientZeroLengthPacket(t *testing.T) {
 		c.Close()
 	}
 }
+
+func TestClientShortPacket(t *testing.T) {
+	// init packet too short.
+	packet := []byte{0, 0, 0, 1, 2}
+
+	r := bytes.NewReader(packet)
+	_, err := NewClientPipe(r, &sink{})
+	if !errors.Is(err, errShortPacket) {
+		t.Fatalf("expected error: %v, got: %v", errShortPacket, err)
```

Comment:
require.ErrorIs?

Reply:
should be ok, now thank you

Reply:
🤷‍♀️ the benefits of using of the `require` package here was always really low.

I know everyone wants to use them, because everyone else does, but you always have to weigh cost/benefits. We’re saving just two lines of code, one of which is just a `}`, and the other is offset by including the `require` package in the first place.

And this `require.ErrorIs` just obscures the `errors.Is()` call from regexps, etc as well.

I’ve had to do significant cleanups on the testing codebase before, because there were a ton of `assert` instead of `require`, where if we just wrote things out clearly, we never would have had that issue. `t.Fatal("unexpected error", err)` is easy to see what it does and how, while `require.NoError(t, err)` vs `assert.NoError(t, err)` does not make it clear which one should be properly used.

If a set of test code is already using that package, then 🤷‍♀️ I’m not going to break convention and start using it. But there’s no particularly compelling reason why we need to drag it newly into every single test file just to save one or two lines.